src/cmd/cmd.cpp

```diff
@@ -167,11 +167,40 @@ void sync(const SyncCTX &ctx)
             }
         }
     });
+    QObject::connect(engine, &SyncEngine::aboutToRemoveAllFiles, engine, [ctx](OCC::SyncFileItem::Direction dir, std::function<void(bool)> callback) {
+        if (!ctx.options.interactive) {
+            callback(false);
+        } else {
+            std::cout << (dir == SyncFileItem::Down ? "All files in the sync folder '%1' folder were deleted on the server.\n"
+                                                      "These deletes will be synchronized to your local sync folder, making such files "
+                                                      "unavailable unless you have a right to restore. \n"
+                                                      "If you decide to keep the files, they will be re-synced with the server if you have rights to do so.\n"
+                                                      "If you decide to delete the files, they will be unavailable to you, unless you are the owner."
+                                                    : "All the files in your local sync folder '%1' were deleted. These deletes will be "
+                                                      "synchronized with your server, making such files unavailable unless restored.\n"
+                                                      "Are you sure you want to sync those actions with the server?\n"
+                                                      "If this was an accident and you decide to keep your files, they will be re-synced from the server.")
+                      << std::endl;
+            std::string s;
+            while (true) {
+                std::cout << "Remove all files?[y,n]";
+                std::getline(std::cin, s);
+                if (s.empty() || s == "y") {
```

Comment:
As we are talking about removing files, should we maybe not accept empty as `y`?